### PR TITLE
Fix foreground state emissions (DEV)

### PR DESCRIPTION
**Pull request by d4rken**
While working on the statistics I noticed that the foreground detection just wasn't working.
We did not emit the values and relied on setting the static property.

### CODE REVIEW

**Review by ralfgehrer — APPROVED** [review]
LGTM.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/ProcessLifecycle.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package de.rki.coronawarnapp.util.di
+
+import javax.inject.Qualifier
+
+@Qualifier
+@MustBeDocumented
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ProcessLifecycle
```
Comment (chris-cwa) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/ProcessLifecycle.kt:8]: why do we need this?

Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/ProcessLifecycle.kt:8]: `ProcressLifecycleOwner.get(): LifecycleOwner`, the interface is generic and there are multiple types of `LifecycleOwner`, so they don't get mixed up in the DI graph, we qualify this specific type of `LifecycleOwner` similar to how we handle `Context`.

Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/ProcessLifecycle.kt:8]:
```kotlin
class Example constructor(@ProcessLifecycle val lifecycleOwner: LifecycleOwner)
```

**Review by chris-cwa — APPROVED** [review]

**Pull request merged**